packages/block-editor/src/hooks/block-style-variation.js

```diff
@@ -59,6 +59,72 @@ function getVariationNameFromClass( className, registeredStyles = [] ) {
 	return null;
 }
 
+export function useGetBlockVariationStylesWithConfig(
```

Comment:
This is a first draft, but the basic logic is to grab the current overrides and update their CSS values with any incoming config. In our case, the user config from a styles revisions.

Ignore the naming. Maybe `useUpdateBlockVariationOverridesWithConfig`

I'm also looking for opportunities to reuse some of the logic within the block style variation hook, but that's a later thing.